### PR TITLE
fix(ses): add status and diagnosticCode for bounce

### LINKED ISSUE #834: Logging of SMTP DSN for bounces
We currently don't log the DSN information for a bounce. It is part of the `bouncedRecipients` object. Having the `status`, `diagnosticCode` values in the logs would help with learning why something bounced.

````
   {
       "notificationType":"Bounce",
       "bounce":{
          "bounceType":"Permanent",
          "reportingMTA":"dns; email.example.com",
          "bouncedRecipients":[
             {
                "emailAddress":"username@example.com",
                "status":"5.1.1",
                "action":"failed",
                "diagnosticCode":"smtp; 550 5.1.1 <username@example.com>... User"
             }
          ],
          "bounceSubType":"General",
          "timestamp":"2012-06-19T01:07:52.000Z",
          "feedbackId":"00000138111222aa-33322211-cccc-cccc-cccc-ddddaaaa068a-000000"
       },
       "mail":{
          "timestamp":"2012-06-19T01:05:45.000Z",
          "source":"sender@example.com",
          "messageId":"00000138111222aa-33322211-cccc-cccc-cccc-ddddaaaa0680-000000",
          "destination":[
             "username@example.com"
          ]
       }
    }
````

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #834

### CODE REVIEW

**Comment by vladikoff** [comment]
@vbudhram @jrgm r?

**Comment by seanmonstar** [comment]
code r+

File: lib/bounces.js [hunk 1]
```diff
@@ -59,7 +59,13 @@ module.exports = function (log, error) {
       }
       return P.each(recipients, function (recipient) {
         var email = recipient.emailAddress
-        log.info({ op: 'handleBounce', email: email, bounce: !!message.bounce })
+        log.info({
+          op: 'handleBounce',
+          email: email,
+          bounce: !!message.bounce,
```
Comment (vbudhram) [inline lib/bounces.js]: One minor thing that I noticed is that you don't log the `bounce.bounceType`? This worthwhile?

File: lib/bounces.js [hunk 2]
```diff
@@ -59,7 +59,13 @@ module.exports = function (log, error) {
...
+          diagnosticCode: recipient.diagnosticCode,
+          status: recipient.status
```
Comment (vbudhram) [inline lib/bounces.js]: According to docs, look like there is a `action` value in the Bounce Object? Worthwhile to add?

**Comment by vbudhram** [comment]
@vladikoff Just read title and you were only adding `status and diagnosticCode`, so up to you on nits, LGTM 👍

**Comment by vladikoff** [comment]
Thanks for reviews @vbudhram and @seanmonstar !

**Pull request merged**